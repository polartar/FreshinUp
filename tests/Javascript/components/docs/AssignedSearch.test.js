import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/docs/AssignedSearch.vue'

describe('AssignedSearch', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          type: 2,
          onAssignChange: () => {},
          onTypeChange: () => {}
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })

    test('selectAssigned function emitted assign-change', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.selectAssigned({ uuid: 'mock' })
      wrapper.vm.selectAssigned()
      expect(wrapper.emitted()['assign-change']).toBeTruthy()
      expect(wrapper.emitted()['assign-change'][1][0]).toEqual('')
    })
  })

  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('currentOption', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          type: 2
        }
      })
      expect(wrapper.vm.currentOption.value).toBe(2)
    })

    test('typeValue change different', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          type: 1
        }
      })

      wrapper.vm.type = 2
      expect(wrapper.vm.typeValue).toBe(2)
      wrapper.vm.typeValue = 3
      expect(wrapper.emitted()['type-change']).toBeTruthy()
      expect(wrapper.emitted()['type-change'][0][0]).toEqual(3)
    })

    test('typeValue change same', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          type: 1
        }
      })

      wrapper.vm.typeValue = 1
      expect(wrapper.emitted()['type-change']).toBeFalsy()
    })
  })
})
