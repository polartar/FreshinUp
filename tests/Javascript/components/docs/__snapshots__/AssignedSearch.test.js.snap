// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignedSearch Snapshots defaults 1`] = `
<div
  class="layout row wrap justify-space-between"
  onassignchange="() => {}"
  ontypechange="() => {}"
>
  <div
    class="flex md5 sm12"
  >
    <div
      class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select v-input--is-label-active v-input--is-dirty theme--light"
      role="combobox"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-select__slot"
          >
            <div
              class="v-select__selections"
            >
              <div
                class="v-select__selection v-select__selection--comma"
              >
                Fleet Member
              </div>
              <input
                aria-readonly="false"
                autocomplete="on"
                data-vv-name="type"
                readonly="readonly"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light "
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          User
                        </div>
                      </div>
                    </a>
                  </div>
                  <div
                    class="primary--text"
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          Fleet Member
                        </div>
                      </div>
                    </a>
                  </div>
                  <div
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          Venue
                        </div>
                      </div>
                    </a>
                  </div>
                  <div
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          Event
                        </div>
                      </div>
                    </a>
                  </div>
                  <div
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          Event/Fleet Memeber
                        </div>
                      </div>
                    </a>
                  </div>
                  <div
                    role="listitem"
                  >
                    <a
                      class="v-list__tile v-list__tile--link theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          Event/Venue
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <div
              class="v-messages__wrapper"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex md6 sm12"
  >
    <div
      background-color="white"
      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      height="48"
      onassignchange="() => {}"
      ontypechange="() => {}"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot white"
          style="height: 48px;"
        >
          <div
            class="v-select__slot"
          >
            <input
              aria-label=""
              autocomplete="off"
              onassignchange="() => {}"
              ontypechange="() => {}"
              placeholder="Search Fleet Member"
              role="combobox"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
